common/http.go

```diff
@@ -210,8 +210,9 @@ func (c *DatabricksClient) checkHTTPRetry(ctx context.Context, resp *http.Respon
 
 // Get on path
 func (c *DatabricksClient) Get(path string, request interface{}, response interface{}) error {
-	if c.authVisitor == nil {
-		return fmt.Errorf("Authentication not initialized")
+	err := c.ensureAuthenticationIsSetup()
```

Comment:
1) create method `authenticatedQuery` with same signature as `genericQuery` 
2) do error check there
3) replace `genericQuery` with `authenticatedQuery` in `Get/Post/Put/Patch/Delete/Scim/OldAPI`, but don't pass in authVisitor, because it should be handled in `authenticatedQuery`.
4) this is happening from multiple threads, so add a mutex as field of `DatabricksClient` struct, that would be managed within either `authenticatedQuery` or `findAndApplyAuthorizer`. otherwise even more unexplainable bugs will happen

this way we'll have less copy-pasted lines and single place of control.